Rewrite all documentation (#1936)
openhab/openhabian: docs/openhabian-troubleshooting.md

Your system still has limits and they will be easily reached if you get a bunch of things going to be conservative.

Additionally, running a 64 bit image on only 1 GB of RAM tends to result in the same situation as running on a system with [less than 1 GB of RAM](#less-than-1-gb-of-ram) so probably don't try to do that either.

#### 2+ GB of RAM

You probably have enough RAM to run all the packages you want unless you are going crazy so you should probably keep reading to see what else could be going wrong.

### Network Access

openHABian requires direct Internet access for the duration of the installation.
Ensure that your openHABian box has access to the internet and is not being blocked by any settings configured in your router.
Both Ethernet and Wi-Fi are both supported at install time.

If none of the things listed below work you are going to need to do more research on your own.
Networking is highly complex and there are many different things that could be the issue.

#### Wi-Fi

WiFi requires user configuration prior to the first boot of openHABian.
For more information on how to configure Wi-Fi before first boot see [Wi-Fi Settings](./openhabian.md#wi-fi-settings)

> If you would rather not try to do any additional configuration before first boot you can try to make use of the [hotspot](./openhabian.md#wi-fi-hotspot) feature.

Author: Somewhat, but it is also partially because much of that content has been shifted around and moved to the main document and this references back to it.
Especially because much of it was a restatement of what was in the main install docs and has been clarified there.

I also removed a little bit of it because some of it was beyond the scope of what we really support anyways.

> If you would rather not try to do any additional configuration before first boot you can try to make use of the [hotspot](./openhabian.md#wi-fi-hotspot) feature.

Reviewer: > but it is also partially because much of that content has been shifted around and moved to the main document

The DEBUG guide was meant to be a standalone document to point people with problems at, and specifically also those that ignored the official docs in the first place. I agree we should rewrite it but I also think we should stick to that idea.
It was written back when we did not have the hotspot and we did not have network manager.
Both have implications that we should explain a lot better, eventually including what you can do on the comitup console if it does not come up in the first place, or Network manager commands rather than `ifconfig`s
Plus mention static IPs prominently (that you must not attempt that and what to do to nonetheless obtain the same IP on every DHCP).

There is not one thing you can check so be aware of that.

#### 1 GB of RAM

For systems with 1 GB of RAM then you will probably be fine with most things and even be able to run a couple of extra components.
Your system still has limits and they will be easily reached if you get a bunch of things going to be conservative.

Additionally, running a 64 bit image on only 1 GB of RAM tends to result in the same situation as running on a system with [less than 1 GB of RAM](#less-than-1-gb-of-ram) so probably don't try to do that either.

If you are running an older 32 bit system on 1 GB of RAM is typically fine.
Be aware that when openHAB 5 releases, 32 bit support will be dropped so you won't be able to upgrade to openHAB 5 without switching over to a 64 bit system.

#### 2+ GB of RAM

You probably have enough RAM to run all the packages you want unless you are going crazy so you should probably keep reading to see what else could be going wrong.

### Network Access

openHABian requires direct Internet access for the duration of the installation.
Ensure that your openHABian box has access to the internet and is not being blocked by any settings configured in your router.

> Both Ethernet and Wi-Fi are both supported at install time.

Reviewer: ```suggestion
Both Ethernet and Wi-Fi are supported at install time.
```

> 

Reviewer: ```suggestion

openHABian will use Ethernet if connected.
If neither Ethernet nor [static Wi-Fi configuration](#Wi-Fi) work and the Internet cannot be reached, the installer will fire up a hotspot with a wireless LAN named `openhabian-<n>`.
Use your mobile to connect to it, to select your home Wi-Fi or to enter its name and openHABian will connect there permanently, i.e. safe across reboots.
```

If none of the things listed below work you are going to need to do more research on your own.
Networking is highly complex and there are many different things that could be the issue.

#### Wi-Fi

> WiFi requires user configuration prior to the first boot of openHABian.

Reviewer: ```suggestion
A non-interactive installation with static use of your home Wi-Fi requires user configuration action prior to the first boot of openHABian.
```
